### PR TITLE
Add dependency mapping RFC addendum

### PR DESCRIPTION
**Pull request by sophiewigmore**
## Summary

Related to resolving https://github.com/paketo-buildpacks/packit/issues/420
[Readable](https://github.com/paketo-buildpacks/rfcs/blob/defefba470985bf2651a6de041939b61cdeb1a1a/text/0010-dependency-mappings.md)

## Checklist

* [X] I have viewed, signed, and submitted the Contributor License Agreement.
* [X] I have linked issue(s) that this PR should close using keywords or the Github UI (See [docs](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue))
* [X] I have added an integration test, if necessary.
* [X] I have reviewed the [styleguide](https://github.com/paketo-buildpacks/community/blob/main/STYLEGUIDE.md) for guidance on my code quality.
* [X] I'm happy with the commit history on this PR (I have rebased/squashed as needed).

### CODE REVIEW

**Comment by ryanmoran** [comment]
I don't think this needs to be its own file. I think you can modify the existing RFC with this addendum content. Should make it easier to find.

**Comment by ryanmoran** [comment]
Should the addendum show that the old format is still supported? We don't mean to make the original SHA256-only format non-functional with this change, right?

**Comment by sophiewigmore** [comment]
That's what this line is meant to explain:
```
The dependency mapping implementation in the project will continue to support the use of a sha256 as long as the sha256 buildpack.toml field is supported; however, users of this feature should ideally use a checksum.
```
I can make this clearer  though with an example.

File: text/0010-dependency-mappings.md [hunk 1]
```diff
@@ -11,10 +11,26 @@ Consumers of Paketo buildpacks may be building in an environment where certain d
 ## Detailed Explanation
 
 ### Discovery
-If one or more bindings of kind or type `dependency-mapping` are present, buildpacks should respect any URIs found in the binding. The `sha256` of the dependency should be used to lookup mapped URIs.
+If one or more bindings of kind or type `dependency-mapping` are present, buildpacks should respect any URIs found in the binding. The lookup digest should be included as `checksum` in the form of `<algorithm>:<hash>` (or `sha256` in the form of `<hash>`) of the dependency should be used to lookup mapped URIs.
```
Comment (dmikusa) [inline text/0010-dependency-mappings.md:14]:
```suggestion
If one or more bindings of kind or type `dependency-mapping` are present, then a buildpack should use the bindings in its dependency resolution process. The dependency's digest, included as `checksum` in the form of `<algorithm>:<hash>` (or `sha256` in the form of `<hash>`), should be used to lookup an alternative URIs.
```

Splitting hairs, but I think this sentence is a little confusing. Suggestion ^^.

File: text/0010-dependency-mappings.md [hunk 2]
```diff
@@ -51,4 +76,7 @@ Before downloading a dependency the buildpack searches all bindings of type `dep
 ## Prior Art
 
 
-## Unresolved Questions and Bikeshedding
\ No newline at end of file
+## Unresolved Questions and Bikeshedding
+
+## Addendums
+- November 3, 2022: Modify RFC to leverage `checksum` for digest as well as a `sha256`
```
Comment (dmikusa) [inline text/0010-dependency-mappings.md:82]:
```suggestion
- November 3, 2022: Modify RFC to leverage `checksum` for digest as well as a `sha256`
- The CNB binding specification is deprecated and while still supported, you should use the Service Binding specification for Kubernetes instead.
```

Slightly off-topic, but perhaps a good time to add this addendum as well.

File: text/0010-dependency-mappings.md [hunk 3]
```diff
@@ -11,10 +11,26 @@ Consumers of Paketo buildpacks may be building in an environment where certain d
...
 
 ### Discovery
-If one or more bindings of kind or type `dependency-mapping` are present, buildpacks should respect any URIs found in the binding. The `sha256` of the dependency should be used to lookup mapped URIs.
+If one or more bindings of kind or type `dependency-mapping` are present, buildpacks should respect any URIs found in the binding. The lookup digest should be included as `checksum` in the form of `<algorithm>:<hash>` (or `sha256` in the form of `<hash>`) of the dependency should be used to lookup mapped URIs.
+
+If the digest is provided in the form of `<hash>`, it will be assumed to be of algorithm type `sha256`.
+
+The dependency mapping implementation in the project will continue to support the use of a `sha256` as long as the `sha256` `buildpack.toml` field is supported; however, users of this feature should ideally use a `checksum`.
 
 ### CNB Bindings Specification Example
```
Comment (dmikusa) [inline text/0010-dependency-mappings.md:20]: Again, splitting hairs, but I think we should put this specification second and make a note that it's deprecated and we suggest folks use the Service Binding spec for K8s.

File: text/0010-dependency-mappings.md (hunk 1 above)
Comment (sophiewigmore) [reply text/0010-dependency-mappings.md:14]: made a slight change, incorporating it now.

**Review by dmikusa — APPROVED** [review]

**Review by ryanmoran — APPROVED** [review]

**Comment by davidmirror-ops** [comment]
WG 11-15-2022: OK to merge

**Pull request merged**